src/golang/cmd/server/handler/create_table.go

```diff
@@ -0,0 +1,178 @@
+	aq_context "github.com/aqueducthq/aqueduct/lib/context"
+	"github.com/aqueducthq/aqueduct/lib/database"
+	"github.com/aqueducthq/aqueduct/lib/job"
+	"github.com/aqueducthq/aqueduct/lib/storage"
+	"github.com/aqueducthq/aqueduct/lib/vault"
+	"github.com/aqueducthq/aqueduct/lib/workflow/operator/connector"
+	"github.com/aqueducthq/aqueduct/lib/workflow/operator/connector/auth"
+	"github.com/aqueducthq/aqueduct/lib/workflow/utils"
+	"github.com/dropbox/godropbox/errors"
+	"github.com/go-chi/chi"
+	"github.com/google/uuid"
+)
+
+// Route: /integration/{integrationId}/create
+// Method: POST
+// Params:
+//	`integrationId`: ID for `integration` object
+//	** ONLY SUPPORTS CREATING TABLES FOR THE DEMO DB **
+// Request:
+//	Headers:
+//		`table-name`: name of table to create
+//		`api-key`: user's API Key
+// Body:
+//		the CSV file to upload to the integration.
+
+const (
+	pollCreateInterval = 500 * time.Millisecond
+	pollCreateTimeout  = 2 * time.Minute
+)
+
+// Creates a table in the specified integration.
+type CreateTableHandler struct {
+	PostHandler
+
+	Database          database.Database
+	IntegrationReader integration.Reader
+	StorageConfig     *shared.StorageConfig
+	JobManager        job.JobManager
+	Vault             vault.Vault
+}
+
+type CreateTableArgs struct {
+	*aq_context.AqContext
+	tableName     string
+	integrationId uuid.UUID
+	csv           []byte
+}
+
+type CreateTableResponse struct{}
+
+func (*CreateTableHandler) Name() string {
+	return "CreateTable"
+}
+
+func (h *CreateTableHandler) Prepare(r *http.Request) (interface{}, int, error) {
+	aqContext, statusCode, err := aq_context.ParseAqContext(r.Context())
+	if err != nil {
+		return nil, statusCode, errors.Wrap(err, "Unable to parse arguments.")
+	}
+	tableName := r.Header.Get(routes.TableNameHeader)
+	integrationIdStr := chi.URLParam(r, routes.IntegrationIdUrlParam)
+	integrationId, err := uuid.Parse(integrationIdStr)
+	if err != nil {
+		return nil, http.StatusBadRequest, errors.Wrap(err, "Malformed integration ID.")
+	}
+
+	csv, err := io.ReadAll(r.Body)
+	if err != nil {
+		return nil, http.StatusBadRequest, errors.Wrap(err, "Unable to read CSV content.")
+	}
+
+	return &CreateTableArgs{
+		AqContext:     aqContext,
+		tableName:     tableName,
+		integrationId: integrationId,
+		csv:           csv,
+	}, http.StatusOK, nil
+}
+
+func (h *CreateTableHandler) Perform(ctx context.Context, interfaceArgs interface{}) (interface{}, int, error) {
+	args := interfaceArgs.(*CreateTableArgs)
+
+	integrationObject, err := h.IntegrationReader.GetIntegration(
+		ctx,
+		args.integrationId,
+		h.Database,
+	)
+	if err != nil {
+		return nil, http.StatusBadRequest, errors.Wrap(err, "Cannot get integration.")
+	}
+
+	// Save CSV
+	contentPath := fmt.Sprintf("create-table-content-%s", args.RequestId)
+	if err := storage.NewStorage(h.StorageConfig).Put(ctx, contentPath, args.csv); err != nil {
+		return nil, http.StatusInternalServerError, errors.Wrap(err, "Cannot save CSV.")
+	}
```

Comment:
let's add a `defer storage.Delete(path)` to always delete the file once the upload is completed